File: 0350-intersection-of-two-arrays-ii-两个数组的交集 II/0350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> um;
        vector<int> res;
        for (auto &a : nums1) {
            um[a]++;
        }
        for (auto &a : nums2) {
            if (um.count(a)) {
                if (um[a] > 0) {
                    um[a]--;
                    res.push_back(a);
                }
            }
        }
        return res;
    }
};

File: 0350-intersection-of-two-arrays-ii-两个数组的交集II/0350.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        list1 = {}
        ans = []
        
        for n1 in nums1:
            if n1 in list1:
                list1[n1] += 1
            else:
                list1[n1] = 1
        
        for n2 in nums2:
            if n2 in list1:
                if list1[n2] > 0:
                    list1[n2] -= 1
                    ans.append(n2)

        return ans